samples/basic-order-processing-client-nestjs/src/use-cases/create-order.use-case.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable, Logger } from "@nestjs/common";
+import { AmqpClientService } from "@amqp-contract/client-nestjs";
+import type { orderContract } from "@amqp-contract-samples/basic-order-processing-contract";
+
+export interface CreateOrderInput {
+  orderId: string;
+  customerId: string;
+  items: Array<{ productId: string; quantity: number; price: number }>;
+  totalAmount: number;
+}
+
+@Injectable()
+export class CreateOrderUseCase {
+  private readonly logger = new Logger(CreateOrderUseCase.name);
+
+  constructor(private readonly amqpClient: AmqpClientService<typeof orderContract>) {}
+
+  async execute(order: CreateOrderInput) {
```

Comment:
public method should always declare their return type. enforce this rule using correct tsconfig in every projects